src/Tests/dotnet-workload-install.Tests/GivenFileBasedWorkloadInstall.cs

```diff
@@ -26,6 +27,46 @@ public GivenFileBasedWorkloadInstall(ITestOutputHelper log) : base(log)
             _manifestPath = Path.Combine(_testAssetsManager.GetAndValidateTestProjectDirectory("SampleManifest"), "Sample2.json");
         }
 
+        [Fact]
+        public void InstallStateUpdatesWorkProperly()
+        {
+            (string dotnetRoot, FileBasedInstaller installer, _, _) = GetTestInstaller();
+            var stringFeatureBand = "6.0.300"; // This is hard-coded in the test installer, so if that changes, update this, too.
+            var sdkFeatureBand = new SdkFeatureBand(stringFeatureBand);
+            var path = Path.Combine(dotnetRoot, "metadata", "workloads", stringFeatureBand, "InstallState", "default.json");
+
+            // The install state shouldn't exist, but delete it in case it wasn't cleaned up properly from the last run.
+            if (File.Exists(path))
+            {
+                File.Delete(path);
+            }
+
+            installer.UpdateInstallMode(sdkFeatureBand, true);
+            var installState = InstallStateContents.FromString(File.ReadAllText(path));
+            installState.manifests.Should().BeNull();
+            installState.useWorkloadSets.Should().BeTrue();
+
+            installer.SaveInstallStateManifestVersions(sdkFeatureBand, new Dictionary<string, string>()
+            {
+                { "first", "second" },
+                { "third", "fourth" },
+            });
+            installState = InstallStateContents.FromString(File.ReadAllText(path));
+            installState.manifests.Count.Should().Be(2);
+            installState.manifests["first"].Should().Be("second");
+            installState.manifests["third"].Should().Be("fourth");
+            installState.useWorkloadSets.Should().BeTrue();
+
+            installer.UpdateInstallMode(sdkFeatureBand, false);
+            installState = InstallStateContents.FromString(File.ReadAllText(path));
```

Comment:
I would add a check here to be sure it still preserved the manifest versions.

```suggestion
            installState = InstallStateContents.FromString(File.ReadAllText(path));
            installState.manifests.Count.Should().Be(2);
```